napari/_qt/qt_plugin_report.py

```diff
@@ -135,8 +136,12 @@ def set_plugin(self, plugin: str) -> None:
         plugin : str
             name of a plugin that has created an error this session.
         """
+        try:
+            self.github_button.clicked.disconnect()
+        except (RuntimeError, TypeError):
+            pass
```

Comment:
@tlambert03 can you add a comment about why this is needed? it's very mysterious.

Reply:
I see also that it's repeated below... Are both actually needed?

Reply:
sure, disconnecting a signal that has no connected slots is an error... but this seems to be the best way to make sure you have nothing connected.  (It's a runtime error in pyside, and type error on pyqt5)

Reply:
where is it repeated?

Reply:
Lines 153-158:

https://github.com/napari/napari/pull/1194/files#diff-9164dcf0ab67f9870f138832a24fdc1dR153-R158

Reply:
(They are not changed lines, they are already there.)

Reply:
In the comment, I meant, note down that we are disconnecting any *previous* plugin's GH button click before connecting to this plugin's GH button click.

Reply:
lol.  sorry I missed that.  and that one has a comment!   will remove

Reply:
The fact that the disconnect and the reconnect are separated in the function makes it confusing, so if they can happen together, that would be ideal.

Reply:
I removed the duplicate, but it's hard to put them together.  We always have to disconnect the last callback when we change the plugin we're viewing (so it makes sense to do that at the top) but we only reconnect it if the new plugin has a github link.

Reply:
@tlambert03 

> I meant, note down that we are disconnecting any previous plugin's GH button click before connecting to this plugin's GH button click.

Could you add a comment right before the disconnection that this is what's happening? Other than that this is :+1: from me!

Reply:
Actually, scratch that, life's too short. =P